agent/exec/container/container.go

```diff
@@ -187,51 +171,80 @@ func (c *containerConfig) tmpfs() map[string]string {
 	return r
 }
 
-func (c *containerConfig) binds() []string {
-	var r []string
+func (c *containerConfig) mounts() []enginemount.Mount {
+	var r []enginemount.Mount
 	for _, mount := range c.spec().Mounts {
-		if mount.Type == api.MountTypeBind || (mount.Type == api.MountTypeVolume && mount.Source != "") {
-			spec := fmt.Sprintf("%s:%s", mount.Source, mount.Target)
-			mask := getMountMask(&mount)
-			if mask != "" {
-				spec = fmt.Sprintf("%s:%s", spec, mask)
-			}
-			r = append(r, spec)
-		}
+		r = append(r, convertMount(mount))
 	}
 	return r
 }
 
-func getMountMask(m *api.Mount) string {
-	var maskOpts []string
-	if m.ReadOnly {
-		maskOpts = append(maskOpts, "ro")
+func convertMount(m api.Mount) enginemount.Mount {
+	mount := enginemount.Mount{
+		Source:   m.Source,
+		Target:   m.Target,
+		ReadOnly: m.ReadOnly,
 	}
 
 	switch m.Type {
-	case api.MountTypeVolume:
-		if m.VolumeOptions != nil && m.VolumeOptions.NoCopy {
-			maskOpts = append(maskOpts, "nocopy")
-		}
 	case api.MountTypeBind:
-		if m.BindOptions == nil {
-			break
-		}
+		mount.Type = enginemount.TypeBind
+	case api.MountTypeVolume:
+		mount.Type = enginemount.TypeVolume
+	default:
+		panic("cannot convert type %s")
```

Comment:
Is panicking really the right thing to do here? The spec is user data, so as I understand it we'd be panicing on bad input.